nixos/modules/services/networking/supybot.nix

```diff
@@ -3,54 +3,90 @@
 with lib;
 
 let
-
   cfg  = config.services.supybot;
-
+  isStateDirHome = hasPrefix "/home/" cfg.stateDir;
+  isStateDirVar = cfg.stateDir == "/var/lib/supybot";
+  pyEnv = pkgs.python3.withPackages (p: [ p.limnoria ] ++ (cfg.extraPackages p));
 in
-
 {
-
   options = {
 
     services.supybot = {
 
       enable = mkOption {
+        type = types.bool;
         default = false;
-        description = "Enable Supybot, an IRC bot";
+        description = "Enable Supybot, an IRC bot (also known as Limnoria).";
       };
 
       stateDir = mkOption {
-        # Setting this to /var/lib/supybot caused useradd to fail
-        default = "/home/supybot";
+        type = types.path;
+        default = if versionAtLeast config.system.stateVersion "20.09"
+          then "/var/lib/supybot"
+          else "/home/supybot";
+        defaultText = "/var/lib/supybot";
         description = "The root directory, logs and plugins are stored here";
       };
 
       configFile = mkOption {
         type = types.path;
         description = ''
-          Path to a supybot config file. This can be generated by
+          Path to initial supybot config file. This can be generated by
```

Comment:
Why is this module coded in a way that the configuration file is initial only? :disappointed:

Reply:
Good question! I think that most supybot operators generate initial configuration with `supybot-wizard` and afterwards configure it by sending it commands over IRC. Even [the user guide discourages](http://doc.supybot.aperio.fr/en/latest/use/configuration.html#editing-the-configuration-values-by-hand) editing the config file.

I was considering adding `mutableConfig` option (`true` by default to be backwards compatible) but figured not much people would use it. It's a simple change though, can add it if it makes sense.